makefiles/boards/sam0.inc.mk

```diff
@@ -5,15 +5,17 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
 # DEBUG_ADAPTER_ID="ATML..."
 
 # The SERIAL setting is only available for backwards compatibility with older
-# settings.
-ifneq (,$(SERIAL))
-  EDBG_ARGS += --serial $(SERIAL)
-  SERIAL_TTY = $(firstword $(shell $(RIOTTOOLS)/usb-serial/find-tty.sh $(SERIAL)))
-  ifeq (,$(SERIAL_TTY))
-    $(error Did not find a device with serial $(SERIAL))
+# settings. It's only checked when a target is given to the make command.
+ifneq (,$(filter-out all %clean,$(MAKECMDGOALS)))
```

Comment:
I would do it the other way around, and only enable this for targets that interact with the board `flash, flash-only, term, cleanterm, test, debug`, etc...

Reply:
This was my first version. I went for the sort list but I can update that sure.

Reply:
Done